src/navigation/path.cpp

```diff
@@ -139,12 +148,49 @@ std::vector<glm::dvec3> Path::controlPoints() const {
 CameraPose Path::traversePath(double dt, float speedScale) {
     double speed = speedAlongPath(_traveledDistance);
     speed *= static_cast<double>(speedScale);
-    const double displacement =  dt * speed;
+    double displacement =  dt * speed;
+
+    const double prevDistance = _traveledDistance;
 
     _progressedTime += dt;
     _traveledDistance += displacement;
 
-    return interpolatedPose(_traveledDistance);
+    CameraPose newPose;
+
+    if (_type == Type::Linear) {
+        // Special handling of linear paths, so that it can be used when we are
+        // traversing very large distances without introducing precision problems
+        const glm::dvec3 prevPosToEnd = _prevPose.position - _end.position();
+        const double remainingDistance = glm::length(prevPosToEnd);
+
+        // Actual displacement may not be bigger than remaining distance
+        if (displacement > remainingDistance) {
+            displacement = remainingDistance;
+            _traveledDistance = pathLength();
+            _forceQuit = true;
+            return _end.pose();
+        }
+
+        // Just move along the line from the current position to the target
+        newPose.position = _prevPose.position -
+            displacement * glm::normalize(prevPosToEnd);
+
+        const double relativeDistance = _traveledDistance / pathLength();
+        newPose.rotation = interpolateRotation(relativeDistance);
+    }
+    else {
```

Comment:
maybe better to have an `else if` here and then another `else` that throws a `ghoul::MissingCaseException` to make it easier in the future to find places where we have to add additional checks

Reply:
So, the condition for that `else if` would be..? A check for any of the other curve types?

If so, then a switch statement would probably be more suitable. However, ideally, I wouldn't want to have any other special cases. Right now, the linear paths are a workaround to help with precision problems but later on, I hope that these will be solved (for example by switching the direction of the interpolation, half-way, as we've talked a bit about). So, no other additional checks should be needed here IMO 🤔

Reply:
:+1:  I saw that after I wrote the comment;  I thought it was a binary enumA vs enumB   not enumA vs all other enums